lisa/tools/python.py

```diff
@@ -89,3 +94,79 @@ def exists_package(self, package_name: str) -> bool:
     def uninstall_package(self, package_name: str) -> bool:
         result = self.run(f"uninstall {package_name} -y", force_run=True, sudo=True)
         return result.exit_code == 0
+
+
+class PythonVenv(Tool):
+    @property
+    def command(self) -> str:
+        return f"{self.get_venv_path()}/bin/{self.python.command}"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Python]
+
+    def _create_venv(self, venv_path: str = "", sudo: bool = False) -> PurePath:
+        _venv_path: PurePath = (
+            PurePath(venv_path) if venv_path else self.node.working_path / "venv"
+        )
+        cmd_result = self.python.run(
+            f"-m venv {_venv_path}", force_run=True, sudo=sudo, shell=True
+        )
+        assert_that(
+            cmd_result.exit_code, f"fail to create venv: {_venv_path}"
+        ).is_equal_to(0)
+        self._venv_path = _venv_path
+        return self._venv_path
+
+    def __init__(self, node: "Node", venv_path: str = "") -> None:
+        super().__init__(node)
+        self.python: Python = self.node.tools[Python]
+        self._venv_installation_path = venv_path if venv_path else ""
+
+    def _check_exists(self) -> bool:
+        venv = self.python.run("-m venv --help", force_run=True)
+        ensurepip = self.python.run("-m ensurepip", force_run=True)
+        return (
+            venv.exit_code == 0 and "No module named ensurepip" not in ensurepip.stdout
+        )
+
+    def _install(self) -> bool:
+        if isinstance(self.node.os, Posix):
+            self.node.os.install_packages("python3-venv")
+        return self._check_exists()
+
+    # if get_venv_path is invoked before create_venv, it will create
+    # a venv in the node working path
+    def get_venv_path(self) -> PurePath:
+        if not hasattr(self, "_venv_path"):
+            self._venv_path = self._create_venv(self._venv_installation_path)
+        return self._venv_path
+
+    def install_packages(self, packages_name: str, sudo: bool = False) -> None:
```

Comment:
sudo is not necessary. venv is to avoid install in root permission.

Reply:
removed all usages of sudo